test/core/utils/matches.js

```diff
@@ -132,38 +132,87 @@ describe('utils.matches', function() {
   });
 
   describe('pseudos', function() {
-    it('returns true if :not matches using tag', function() {
+    it('throws error if pseudo is not implemented', function() {
       var virtualNode = queryFixture('<h1 id="target">foo</h1>');
-      assert.isTrue(matches(virtualNode, 'h1:not(span)'));
+      assert.throws(function() {
+        matches(virtualNode, 'h1:empty');
+      });
+      assert.throws(function() {
+        matches(virtualNode, 'h1::before');
+      });
     });
 
-    it('returns true if :not matches using class', function() {
-      var virtualNode = queryFixture('<h1 id="target">foo</h1>');
-      assert.isTrue(matches(virtualNode, 'h1:not(.foo)'));
-    });
+    describe(':not', function() {
+      it('returns true if :not matches using tag', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isTrue(matches(virtualNode, 'h1:not(span)'));
+      });
 
-    it('returns true if :not matches using attribute', function() {
-      var virtualNode = queryFixture('<h1 id="target">foo</h1>');
-      assert.isTrue(matches(virtualNode, 'h1:not([class])'));
-    });
+      it('returns true if :not matches using class', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isTrue(matches(virtualNode, 'h1:not(.foo)'));
+      });
 
-    it('returns true if :not matches using id', function() {
-      var virtualNode = queryFixture('<h1 id="target">foo</h1>');
-      assert.isTrue(matches(virtualNode, 'h1:not(#foo)'));
-    });
+      it('returns true if :not matches using attribute', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isTrue(matches(virtualNode, 'h1:not([class])'));
+      });
 
-    it('returns false if :not matches element', function() {
-      var virtualNode = queryFixture('<h1 id="target">foo</h1>');
-      assert.isFalse(matches(virtualNode, 'h1:not([id])'));
+      it('returns true if :not matches using id', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isTrue(matches(virtualNode, 'h1:not(#foo)'));
+      });
+
+      it('returns true if :not matches none of the selectors', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isTrue(matches(virtualNode, 'h1:not([role=heading], span)'));
+      });
+
+      it('returns false if :not matches element', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isFalse(matches(virtualNode, 'h1:not([id])'));
+      });
+
+      it('returns false if :not matches one of the selectors', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isFalse(matches(virtualNode, 'h1:not([role=heading], h1)'));
```

Comment:
You want an h1 that's not an h1? xD How about:

```suggestion
        assert.isFalse(matches(virtualNode, 'h1:not([role=heading], [id])'));
```